### DISCUSSION TITLE
Can't add MongoDB integration

### DISCUSSION THREAD

**Discussion by fascinated** [vercel/community#2700] [Help]
### Summary

Trying to add a mongodb integration to my Hobby-level account. Process gets stuck on the "We are setting up your integration. This might take a minute." screen.

<img width="700" alt="Screenshot 2023-06-01 at 14 26 45" src="https://github.com/vercel/community/assets/355551/35ae2ace-4c7b-4f88-bf2a-19493845b2c3">

### Steps to Reproduce

After signing into my MongoDB account, selecting the Atlas organization, and picking the new cluster / project mapping, the process just hangs on the "We are setting up your integration. This might take a minute." screen.

If I go into the MongoDB Atlas integration view, under integrations it shows "There are no current Atlas clusters linked to Vercel". If I click "link" there, nothing seems to change. In Vercel the integration also does not appear.

**Comment by amyegan** [comment]
Hi @fascinated. I just tried this with my Next.js test project and didn't run into this error. It may have been a temporary network glitch, or it could be caused by a difference between our projects. Because I couldn't see the error myself, I have a few questions to help me get a better idea of where to start investigating.

- Do you see any MongoDB related environment variables added in your Vercel project settings?
- When adding the integration, do you see any errors in your browser's dev tools console or network log?
- Was the DB completely new or did was it set up beforehand?

**Comment by fascinated** [comment]
Thanks for trying it out! 

* I don't see any MongoDB env variables in the settings :(
* In the console for the window where the process gets stuck, I see two errors:
403 for https://cloud.mongodb.com/settings/addTempApiKey
500 for https://cloud.mongodb.com/orgs/ALPHANUMERIC_ID1/integrations/vercel/ALPHANUMERIC_ID2/link/ALPHANUMERIC_ID3
* Just creating a brand new DB here. I do have a very old MongoDB MMS/Atlas account, so wondering if this is in some way an issue.

**Comment by fascinated** [comment]
For anyone else running into this, it is relatively easy to just manually set up a connection to a mongodb atlas instance too, so do not fear.

To get nearly the same result as the integration:
1) make a mongodb cloud/atlas account
2) create a free atlas cluster (or a pay-per 1M queries type cluster) in the region where your vercel app is (usually north east US/virginia)
3) create a username/password for an admin-level user and another for writing/reading data. use mongodb's suggested strong passwords (or even something better)
4) allow connections from 0.0.0.0/0 (all ip addresses) to this cluster (this is a bit weird, but exactly why you need to use a strong password)
5) set MONGODB_URI in your project's env variables to "mongodb+srv://app-username:app-password@somehost.name.mongodb.net/"
5a) to admin this instance, use MongoDB Compass https://www.mongodb.com/try/download/compass with that connection string
